File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0,r=height.size()-1;
        int maxi=INT_MIN;
        while(l<r){
            int le=min(height[l],height[r]);
            int b=r-l;
            maxi=max(maxi,le*b);
            if(height[l]<=height[r]){
                l++;
            }else{
                r--;
            }
        }
        return maxi;
        
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        ml=0
        while l<=r:
            le=min(height[l],height[r])
            b=r-l
            ml=max(ml,le*b)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return ml

